task-sdk/src/airflow/sdk/execution_time/context.py

```diff
@@ -176,6 +176,10 @@ def _get_variable(key: str, deserialize_json: bool) -> Any:
         try:
             var_val = secrets_backend.get_variable(key=key)  # type: ignore[assignment]
             if var_val is not None:
+                if deserialize_json:
+                    import json
+
+                    var_val = json.loads(var_val)
```

Comment:
We need secret masking too -- check the same implementation in models.variable -- happy for you to do it in separate PR but let's do it for 3.0.2

https://github.com/apache/airflow/blob/13c6020278d1b4eddfcc10b4544ec2bed451a7d5/airflow-core/src/airflow/models/variable.py#L171-L175

Reply:
Good call. Let me just send it in a follow up.